assets/ca/ca.querytranslator.js

```diff
@@ -0,0 +1,387 @@
+ * conversion) the CA search query syntax and the jQuery query builder hierarchical rule set structure.
+ *
+ * The only range filter that is supported here is `between`.  The Lucene / CA search syntax does not readily support
+ * `greater_than` or `less_than`.
+ */
+(function () {
+	var escapeValue, getTokenList, shiftToken, tokensToRuleSet,
+		assertNextToken, isNextToken, assertCondition, skipWhitespace, assignOperatorAndValue,
+		TOKEN_WORD = 'WORD',
+		TOKEN_LPAREN = 'LPAREN',
+		TOKEN_RPAREN = 'RPAREN',
+		TOKEN_COLON = 'COLON',
+		TOKEN_NEGATION = "NEGATION",
+		TOKEN_WHITESPACE = 'WHITESPACE',
+		REGEX_OPERATOR_MAP = [
+			{
+				regex: /^(.*)$/,
+				operator: 'equal',
+				negatedOperator: 'not_equal'
+			},
+			{
+				regex: /^\((.*)\)$/,
+				operator: 'in',
+				negatedOperator: 'not_in'
+			},
+			{
+				regex: /^\[\s*(.*)\s+TO\s+(.*)\s*]$/,
+				operator: 'between',
+				negatedOperator: 'not_between'
+			},
+			{
+				regex: /^([^\*].*[^\\])\*$/,
+				operator: 'begins_with',
+				negatedOperator: 'not_begins_with'
+			},
+			{
+				regex: /^\*(.*[^\\])\*$/,
+				operator: 'contains',
+				negatedOperator: 'not_contains'
+			},
+			{
+				regex: /^\*(.*(?:\\\*|[^\*]))$/,
+				operator: 'ends_with',
+				negatedOperator: 'not_ends_with'
+			},
+			{
+				regex: /^\*$/,
+				operator: 'is_empty',
+				negatedOperator: 'is_not_empty'
+			}
+		];
+
+	/**
+	 * Escape the user-entered field value.
+	 * @param {String} value
+	 * @returns {String}
+     */
+	escapeValue = function (value) {
+		return value.replace(/([\-\+&\|!\(\)\{}\[\]\^"~\*\?:\\])/, '\\$1');
+	};
+
+	/**
+	 * Convert a set of rules from the jQuery query builder into a CA search query.  Performs the inverse operation to
+	 * `convertSearchQueryToQueryBuilderRules`.
+	 * @param {Object} ruleSet
+	 * @returns {String}
+	 */
+	caUI.convertQueryBuilderRuleSetToSearchQuery = function (ruleSet) {
+		if (ruleSet.condition && ruleSet.rules) {
+			return '(' + $.map(ruleSet.rules, caUI.convertQueryBuilderRuleSetToSearchQuery).join(' ' + ruleSet.condition + ' ') + ')';
+		}
+		if (ruleSet.operator && ruleSet.field) {
+			// Escape value to allow special characters
+			var negation = ruleSet.operator.match(/not_/),
+				prefix = ruleSet.field + (negation ? ':-' : ':');
+			switch (negation ? ruleSet.operator.replace('not_', '') : ruleSet.operator) {
+				case 'equal':
+					return prefix + '"' + escapeValue(ruleSet.value) + '"';
+				case 'in':
+					return prefix + '(' + escapeValue(ruleSet.value) + ')';
+				case 'between':
+					return prefix + '[' + escapeValue(ruleSet.value[0]) + ' TO ' + escapeValue(ruleSet.value[1]) + ']';
+				case 'begins_with':
+					return prefix + '"' + escapeValue(ruleSet.value) + '*"';
+				case 'contains':
+					return prefix + '"*' + escapeValue(ruleSet.value) + '*"';
+				case 'ends_with':
+					return prefix + '"*' + escapeValue(ruleSet.value) + '"';
+				case 'is_empty':
+				case 'is_null':
```

Comment:
`*` matches every value, but I'm not sure how it responds to . Selecting `is_empty` for ca_objects.Author results in ca_objects.Author:* and `is_not_empty` results in ca_objects.Author:-* and it should be the opposite.